### ISSUE TITLE
🌟 Limit jobs executed in parallel

### ISSUE THREAD

**Issue by mkanal** [eclipse-tractusx/item-relationship-service#892]
**As** system responsible 
**I want** to execute jobs executed in parallel 
**so that** the system is resilience by using balanced resources.

## Hints / Details

* Logging and monitoring to track the status and performance of each batch during parallel processing. (e.g. StopWatch logs time metrics in DEBUG level)
* Evaluate the need for adjusting the number of ThreadPools based on system resources or statically in configuration .

![Image](https://github.com/user-attachments/assets/82aee0c8-f176-4122-9efd-fe690739c241)

Source: 
[item-relationship-service/docs/src/uml-diagrams/runtime-view/scaling-job-batch-processing.puml at main · eclipse-tractusx/item-relationship-service](https://github.com/eclipse-tractusx/item-relationship-service/blob/main/docs/src/uml-diagrams/runtime-view/scaling-job-batch-processing.puml)

## Acceptance Criteria
- [ ] Limit of jobs executed in parallel in batch processing is implemented and configurable
- [ ] Default value is set to the current value or unlimited if there is no default. 
- [ ] Documentation of the setup and configuration the value 
- [ ] Value is configurable over the HELM Charts 

## Out of Scope

- ...

**Comment by ther3sa** [comment]
Note from Release Management ([@ther3sa](https://github.com/ther3sa)) and Tractus-X Project Lead ([@stephanbcbauer](https://github.com/stephanbcbauer))

Based on the overarching [release check issue](https://github.com/eclipse-tractusx/sig-release/issues/1144), we assume that this feature has been fully implemented. Consequently, we plan to close this feature on Friday.
If this feature has not been implemented, please provide a brief explanation so that we can keep it open accordingly.
Thank you!

**Comment by ther3sa** [comment]
Note from Release Management ([@ther3sa](https://github.com/ther3sa)) and Tractus-X Project Lead ([@stephanbcbauer](https://github.com/stephanbcbauer))

Based on the previous comment, we assume that this feature has been implemented and will close this feature.
